app/models/user.js

```diff
@@ -359,14 +372,28 @@ exports.addModel = function(dbAdapter) {
       hasChanges = true
     }
 
+    if (params.hasOwnProperty('frontendPreferences')) {
+      this.frontendPreferences = params.frontendPreferences
+      hasChanges = true
+    }
+
     if (hasChanges) {
       this.updatedAt = new Date().getTime()
 
+      // Validate maximum preferences store size
+      var prefsString = JSON.stringify(this.frontendPreferences)
+      var prefsLen = GraphemeBreaker.countBreaks(prefsString)
```

Comment:
No, we are talking about front-end preferences which often deal with utf8 strings. We are not much concerned with size in kilobytes (hello, 21st century), more with general limit on the size.